### ISSUE TITLE
Internal Server Error(code 500) returned by apache

### ISSUE THREAD

**Issue by oguya** [oguya/ansible-ckan#6]
Deploying ckan with vanilla configs on Ubuntu 12.04 installs & configures all the components successfully.
But when you access the default ckan URL - http://hostname/ you get a HTTP code 500 error from Apache.

![500 internal server error - google chrome_001](https://cloud.githubusercontent.com/assets/4123209/9244249/3c7d6750-41a2-11e5-8e04-5a87447045b5.png)
- apache version
  
  ```
  root@ubuntu-VirtualBox:~# apachectl -v
  Server version: Apache/2.2.22 (Ubuntu)
  Server built:   Jul 24 2015 17:25:52
  ```
- OS type & version
  
  ```
  root@ubuntu-VirtualBox:~# lsb_release -a
  No LSB modules are available.
  Distributor ID: Ubuntu
  Description:    Ubuntu 12.04.5 LTS
  Release:        12.04
  Codename:       precise
  root@ubuntu-VirtualBox:~# 
  ```
- apache error logs
  
  ```
  root@ubuntu-VirtualBox:~# cat /var/log/apache2/ckan.localhost.localdomain.error.log 
  [Thu Aug 13 09:39:47 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /
  [Thu Aug 13 09:39:47 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /favicon.ico, referer: http://172.26.23.225/
  [Thu Aug 13 09:41:33 2015] [crit] [client 172.26.23.225] configuration error:  couldn't perform authentication. AuthType not set!: /
  [Thu Aug 13 09:42:11 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /
  [Thu Aug 13 09:42:11 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /favicon.ico, referer: http://172.26.23.225/
  [Thu Aug 13 09:42:11 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /
  [Thu Aug 13 09:42:11 2015] [crit] [client 172.26.23.23] configuration error:  couldn't perform authentication. AuthType not set!: /favicon.ico, referer: http://172.26.23.225/
  ```

**Comment by scuerda** [comment]
Not sure if you've solved this, but I came up on the exact problem while writing a similar playbook for deploying ckan.

For me, the solution was (so far) changing the following lines in my .conf file

```
<Directory {{ ckan_settings_path }}>
        Order allow,deny
        Allow from all
        Require all granted
</Directory>
```

to 

```
<Directory {{ ckan_settings_path }}>
        Order allow,deny
        Allow from all
        Require all granted
        Satisfy Any
</Directory>
```

The addition of `Satisfy Any` seems to be the key.

**Comment by oguya** [comment]
Basically the config file [000-ckan-default.conf.j2](https://github.com/oguya/ansible-ckan/blob/master/roles/ckan/templates/apache/000-ckan-default.conf.j2) works on Apache 2.4 & above. 

@scuerda 
By default, Apache 2.2 uses `Satisfy All`[1] which ensures all specified criteria must be satisfied before a user is allowed in. On the hand using `Satisfy Any` works but it's less secure.
1. https://httpd.apache.org/docs/2.2/howto/auth.html#satisfy

**Comment by oguya** [comment]
This is issue occurs on only Apache 2.2 & below which is available on Ubuntu 12.04. I don't see why I should create a fix for this, yet Ubuntu 12.04 is almost approaching its EOL & most people use Ubuntu 14.04 anyways ¯\_(ツ)_/¯

**Comment by scuerda** [comment]
Works for me!
